### PR TITLE
Allow per-operation messenger configuration

### PR DESCRIPTION
**Pull request by soyuka**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | na
| License       | MIT
| Doc PR        | na

Allow messenger attribute by operation.

### CODE REVIEW

**Comment by dunglas** [comment]
Indeed we'll have to introduce a `ContextAwareDataPersister`

File: src/Bridge/Symfony/Messenger/DataPersister.php [hunk 1]
```diff
@@ -43,9 +44,17 @@ public function __construct(ResourceMetadataFactoryInterface $resourceMetadataFa
     /**
      * {@inheritdoc}
      */
-    public function supports($data): bool
+    public function supports($data, array $context = []): bool
     {
-        return true === $this->resourceMetadataFactory->create($this->getObjectClass($data))->getAttribute('messenger');
+        $resourceMetadata = $this->resourceMetadataFactory->create($this->getObjectClass($data));
+
+        $operationType = $context['collection_operation_name'] ?? false ? OperationType::COLLECTION : OperationType::ITEM;
```
Comment (alanpoulain) [inline src/Bridge/Symfony/Messenger/DataPersister.php]: Inside the if?

File: tests/Bridge/Symfony/Messenger/DataPersisterTest.php [hunk 2]
```diff
@@ -72,4 +72,14 @@ public function testHandle()
         $dataPersister = new DataPersister($this->prophesize(ResourceMetadataFactoryInterface::class)->reveal(), $messageBus->reveal());
         $this->assertSame('result', $dataPersister->persist($dummy));
     }
+
+    public function testSupportsPerOperation()
```
Comment (alanpoulain) [inline tests/Bridge/Symfony/Messenger/DataPersisterTest.php]:
```suggestion
    public function testSupportsPerOperation(): void
```

Comment (soyuka) [reply tests/Bridge/Symfony/Messenger/DataPersisterTest.php]: since when do we do this :p

Comment (alanpoulain) [reply tests/Bridge/Symfony/Messenger/DataPersisterTest.php]: Since PHPUnit 8? :stuck_out_tongue:

Comment (meyerbaptiste) [reply tests/Bridge/Symfony/Messenger/DataPersisterTest.php]: Only for:
* `PHPUnit\Framework\TestCase::setUpBeforeClass()`,
* `PHPUnit\Framework\TestCase::setUp()`,
* `PHPUnit\Framework\TestCase::assertPreConditions()`,
* `PHPUnit\Framework\TestCase::assertPostConditions()`,
* `PHPUnit\Framework\TestCase::tearDown()`,
* `PHPUnit\Framework\TestCase::tearDownAfterClass()`,
* `PHPUnit\Framework\TestCase::onNotSuccessfulTest()`.
😄

Comment (alanpoulain) [reply tests/Bridge/Symfony/Messenger/DataPersisterTest.php]: Sure but I think it's a good habit to always do it.

**Pull request merged**